homeassistant/components/honeywell/sensor.py

```diff
@@ -0,0 +1,109 @@
+"""Support for Honeywell (US) Total Connect Comfort sensors."""
+from __future__ import annotations
+
+from homeassistant.components.honeywell.const import (
+    DOMAIN,
+    SENSOR_LOCATION_INDOOR,
+    SENSOR_LOCATION_OUTDOOR,
+)
+from homeassistant.components.sensor import STATE_CLASS_MEASUREMENT, SensorEntity
+from homeassistant.const import (
+    DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_TEMPERATURE,
+    PERCENTAGE,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+)
+import homeassistant.helpers.device_registry as dr
+
+
+async def async_setup_entry(hass, config, async_add_entities, discovery_info=None):
+    """Set up the Honeywell thermostat."""
+    data = hass.data[DOMAIN][config.entry_id]
+    sensors = []
+
+    if data.device.current_temperature is not None:
+        sensors.append(
+            HoneywellUSSensor(data, SENSOR_LOCATION_INDOOR, DEVICE_CLASS_TEMPERATURE)
+        )
+    if data.device.current_humidity is not None:
+        sensors.append(
+            HoneywellUSSensor(data, SENSOR_LOCATION_INDOOR, DEVICE_CLASS_HUMIDITY)
+        )
+    if data.device.outdoor_temperature is not None:
+        sensors.append(
+            HoneywellUSSensor(data, SENSOR_LOCATION_OUTDOOR, DEVICE_CLASS_TEMPERATURE)
+        )
+    if data.device.outdoor_humidity is not None:
+        sensors.append(
+            HoneywellUSSensor(data, SENSOR_LOCATION_OUTDOOR, DEVICE_CLASS_HUMIDITY)
+        )
+
+    async_add_entities(sensors)
+
+
+class HoneywellUSSensor(SensorEntity):
```

Comment:
personally I would inherit from HoneywellUSSensor: HoneywellUSSensorTemperature and HoneywellUSSensorHumidity

Reply:
Totally agree. I will implement this as soon as I have a few spare cycles.